tests/conftest.py

```diff
@@ -1,95 +1,193 @@
 import subprocess
-from threading import Thread
+import tempfile
 import time
-import pytest
-import signal
-import psutil
+from threading import Thread
+
+from contextlib import contextmanager
+from wsgidav.wsgidav_app import WsgiDAVApp
+from cheroot import wsgi
 
 __this_dir__ = os.path.join(os.path.abspath(os.path.dirname(__file__)))
 
+webdav_server_host = "127.0.0.1"
+webdav_server_port = 31102
 
-def kill_child_processes(parent_pid, sig=signal.SIGINT):
-    try:
-        parent = psutil.Process(parent_pid)
-        children = parent.children(recursive=True)
-        for process in children:
-            process.send_signal(sig)
-    except psutil.NoSuchProcess:
-        return
 
+def hash_file(filename):
+    """"Returns the SHA-1 hash of the file provided"""
+    h = hashlib.sha1()
 
-@pytest.fixture
-def start_service(pytestconfig):
+    with open(filename, 'rb') as file:
+        chunk = 0
+        while chunk != b'':
+            chunk = file.read(1024**2)
+            h.update(chunk)
 
-    rootdir = pytestconfig.rootdir
+    return h.hexdigest()
 
-    env = copy.deepcopy(dict(os.environ))
-    print(("rootdir", str(rootdir)))
-    env['PYTHONPATH'] = str(rootdir) + ":" + str(rootdir) + "/tests:" + \
-        str(rootdir) + '/bin:' + \
-        __this_dir__ + ":" + os.path.join(__this_dir__, "../bin:") + \
-        env.get('PYTHONPATH', "")
 
-    env['CTADS_DISABLE_ALL_AUTH'] = 'True'
-    env['CTADS_CABUNDLE'] = "cabundle.pem"
+def generate_random_file(filename, size):
+    with open(filename, 'wb') as fout:
+        fout.write(os.urandom(size))
 
-    print(("pythonpath", env['PYTHONPATH']))
 
-    cmd = ["python", "downloadservice/cli.py"]
+@pytest.fixture(scope="session")
+def app():
+    with tempfile.TemporaryDirectory() as tmpdir:
+        from downloadservice.app import app
 
-    print(f"\033[33mcommand: {cmd}\033[0m")
+        os.system(f"openssl req -newkey rsa:2048 -new -nodes -x509 -days 3650 \
+                -keyout {tmpdir}/key.pem -out {tmpdir}/cert.pem -batch")
 
-    p = subprocess.Popen(
-        cmd,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.STDOUT,
-        shell=False,
-        env=env,
-    )
+        app.config.update({
+            "TESTING": True,
+            "CTADS_DISABLE_ALL_AUTH": True,
+            "DEBUG": True,
+            "CTADS_CABUNDLE": f"{tmpdir}/cert.pem",
+            "CTADS_CLIENTCERT": f"{tmpdir}/cert.pem",
+            'CTADS_UPSTREAM_ENDPOINT':
+                f'http://{webdav_server_host}:{str(webdav_server_port)}/',
+            'CTADS_UPSTREAM_BASEPATH': '',
+            "SERVER_NAME": 'app',
+        })
 
-    url_store = [None]
+        yield app
 
-    def follow_output():
-        url_store[0] = None
-        for line in iter(p.stdout):
-            line = line.decode()
 
-            NC = '\033[0m'
-            if 'ERROR' in line:
-                C = '\033[31m'
-            else:
-                C = '\033[34m'
+@contextmanager
+def webdav_server():
```

Comment:
could I propose to call this `test_upstream_webdav_server` or something like that to avoid confusion in the future. Since the other fixture is also in fact a webdav server, among other things.

Reply:
Hum, the other is not a webdav server, it's the `downloadservice`.

Reply:
Yes, but it exposes webdav server as part of it's functionality, does it not? It can be used purely as webdav server, accessed by webdav client, etc?

Reply:
Changed